markdown/opentelemetry-intro/opentelemetry-intro.md

```diff
@@ -0,0 +1,320 @@
+author: Jesse Menning
+summary:
+id: opentelemetry-intro
+tags:
+categories: opentelemetry, nyctaxi
+environments: Web
+status: Draft
+feedback link: <insert github repo or leave empty>
+analytics account: UA-3921398-10
+
+# Using OpenTelemetry to find your event in a haystack
+
+## What you'll learn: Overview
+
+Duration: 0:05:00
+
+
+Positive
+: If you'd like to learn more about OpenTelemetry architecture, but don't care about hands-on stuff, you can find it in the [Solace architecture blogs](https://solace.com/blog/category/architects/).
+
+No business likes an angry user.  That goes double if the user is a taxi driver. The language can be...intense.
+
+As NYC Modern Taxi Company modernizes its operations through event-based integration, things will go wrong.  What's important is that issues are resolved quickly and as painlessly as possible for the end user, so the taxis keep moving.  To help with that, NYC Modern Taxi is adding OpenTelemetry distributed tracing to all of their microservices.  The goal is to trace events at they move through Solace and potentially dozens of microservices.
+
+Instead of plowing through logs, trying to understand what happened, NYC Modern Taxi can instead search through graphs like these that show how the event moved across microservices: 
+![flame-graph](img/flame-graph.png)
+
+
+In this CodeLab you’ll learn about: 
+
+✅ How OpenTelemetry works (the basics, if you'd like more detail there is a blog post here.
+
+✅ How to get the Solace PubSub+ Python API, and set up a basic environment. 
+
+✅ How to include OpenTelemetry tracing in your processes 
+
+✅ How to visualize your event-driven applications with Jaeger
+
+
+![opentelemetry](img/opentelemetry.png)
+
+
+
+## What you need: Prerequisites 
+
+Duration: 0:07:00
+
+
+1. A general understanding of [event-driven architecture (EDA) terms and concepts](https://docs.solace.com/#Messagin). 
+1. A free trial account of Solace PubSub+ Cloud.   Don't have one? [Sign up here.](https://console.solace.cloud/login/new-account)
+1. [Download Jaeger](https://www.jaegertracing.io/download/) for free in the OS of your choosing
+1. [Download Python](https://www.python.org/downloads/) for free in the OS of your choosing
+1. [Download the sample code](https://github.com/jmenning-solace/nycmoderntaxi/tree/master/opentelemetry-intro)   Or use git commands to grab it from [GitHub](https://github.com/jmenning-solace/nycmoderntaxi.git).
+ 
+
+Negative
+: It's important to note that this CodeLab uses a pre-release version of the Solace PubSub+ Python connector. It has the latest features, but may also have some bugs. Please reach out to our [Solace Community](https://solace.community/) if you happen to find one. 
```

Comment:
Can you link this https://solace.community/discussion/336/python-whos-in-for-a-real-treat instead please